core/sdk.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2023 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rond-authz/rond/internal/metrics"
+	"github.com/rond-authz/rond/openapi"
+	"github.com/rond-authz/rond/types"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"github.com/uptrace/bunrouter"
+)
+
+type PolicyResult struct {
+	QueryToProxy []byte
+	Allowed      bool
+}
+
+// Warning: This interface is experimental, and it could change with breaking also in rond patches.
+// Does not use outside this repository until it is not ready.
+type SDK interface {
+	Metrics() metrics.Metrics
+
+	FindEvaluator(logger *logrus.Entry, method, path string) (SDKEvaluator, error)
+	EvaluatorFromConfig(logger *logrus.Entry, config openapi.RondConfig) SDKEvaluator
+}
+
+// Warning: This interface is experimental, and it could change with breaking also in rond patches.
+// Does not use outside this repository until it is not ready.
+type SDKEvaluator interface {
+	Permission() openapi.RondConfig
+	PartialResultsEvaluators() PartialResultsEvaluators
+}
+
+type evaluator struct {
+	rond       rondImpl
+	logger     *logrus.Entry
+	rondConfig openapi.RondConfig
+}
+
+func (e evaluator) Permission() openapi.RondConfig {
+	return e.rondConfig
+}
+
+func (e evaluator) PartialResultsEvaluators() PartialResultsEvaluators {
+	return e.rond.evaluator
+}
+
+type rondImpl struct {
```

Comment:
This struct is the current interface implementation for the rond service right? can we document it?

Reply:
This is the actual sdk implementation, and it's private. In the future, I expect it will be used outside rond when using the sdk